webapp/publisher/views.py

```diff
@@ -87,13 +87,14 @@ def get_account_details():
     # if anything fails, just continue and don't show
     # this section
     try:
-        marketo_user = marketo.get_user(flask_user["email"])
-        marketo_subscribed = marketo.get_newsletter_subscription(
-            marketo_user["id"]
-        )
         subscribed_to_newsletter = False
-        if marketo_subscribed.get("snapcraftnewsletter"):
-            subscribed_to_newsletter = True
+        marketo_user = marketo.get_user(flask_user["email"])
+        if marketo_user:
+            marketo_subscribed = marketo.get_newsletter_subscription(
+                marketo_user["id"]
+            )
+            if marketo_subscribed.get("snapcraftnewsletter"):
```

Comment:
though if you make the change above for the get_newsletter_subscription method, this will also need updating to check that marketo_subscribed exists AND snapcraftnewsletter is in the dict
```
if marketo_subscribed and marketo_subscribed.get("snapcraftnewlsetter"):
```
wdyt @jkfran ?

Reply:
Well, I have seen that `get_newsletter_subscription` return always a dict. So that means this condition will always work. It could be like that.

Reply:
In that case I leave this condition as it is.

Reply:
But you now updated the method `get_newsletter_subscription` to return `None` so now this condition could raise an exception because you can't use `get()` with None. If you want to return `None` then this needs to be like Luke mention before.

Reply:
So be it.

Reply:
Unless @Lukewh or you (@jkfran ) have a strong opinion about returning an empty dict on https://github.com/canonical-web-and-design/snapcraft.io/blob/master/webapp/api/marketo.py#L95 I will go with that.

Reply:
It is fine to me